File: dani-eth/backend/app/schemas/target.py
from pydantic import BaseModel, Field
from typing import Literal


class ScopeModel(BaseModel):
    scan_type: Literal["full", "quick", "custom"] = "quick"
    ports: str = "1-1024"
    categories: list[Literal["ports", "web", "vuln", "os"]] = ["ports"]


class TargetCreate(BaseModel):
    name: str = Field(..., min_length=2, max_length=255)
    target: str = Field(..., description="IP o dominio del objetivo")
    description: str | None = None
    scope: ScopeModel = Field(default_factory=ScopeModel)


class TargetUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    scope: ScopeModel | None = None


class TargetResponse(BaseModel):
    target_id: str
    name: str
    target: str
    description: str | None
    scope: ScopeModel
    created_at: str
    updated_at: str | None = None

File: dani-eth/frontend/src/types/target.ts
export interface Scope {
  scan_type: 'full' | 'quick' | 'custom';
  ports: string;
  categories: Array<'ports' | 'web' | 'vuln' | 'os'>;
}

export interface Target {
  target_id: string;
  name: string;
  target: string;
  description: string | null;
  scope: Scope;
  created_at: string;
  updated_at: string | null;
}

export interface TargetCreate {
  name: string;
  target: string;
  description?: string;
  scope?: Partial<Scope>;
}

export interface TargetUpdate {
  name?: string;
  description?: string;
  scope?: Partial<Scope>;
}